[ADD] 12.0 - pos_cashback_warning

legalsylvain opened the pull request:
This modules extends the Point of sale functionnalies to add a warning if confirming an order generate a cashback.

By default, if the payment lines of an Odoo PoS order generate change, the journal for the change will be the Cash journal.
That is ok if the original payment line is a Cash payment, but if it's a payment by credit card, check, meal voucher, etc. confirming the transaction will generate a cashback.

This module simply add a message in such case.

![pos_cashback_warning](https://user-images.githubusercontent.com/3407482/191934332-4d508891-c920-4136-a1c3-dac88eec6504.png)


**Note** : The cashback is an operation that consists, for a merchant to collect more money in check, credit card, etc. than the amount of the sale and give change in cash.

CC : @ivantodorovich, @carmenbianca, @robinkeunen, @quentinDupont

legalsylvain committed:
[ADD] pos_cashback_warning

ivantodorovich commented on pos_cashback_warning/__manifest__.py:
```diff
@@ -0,0 +1,25 @@
+        'static/src/xml/pos.xml',
+    ],
+    "images": [
+        "static/description/pos_cashback_warning.png",
```
Question: does this `images` key do something? 👀

legalsylvain replied:
AFAIK, for the apps.odoo.com, the images is used to display the image in the appstore.

See V13 doc : https://www.odoo.com/documentation/13.0/developer/reference/addons/module.html?highlight=manifest

but in master, the key seems to be removed : https://www.odoo.com/documentation/15.0/developer/reference/backend/module.html?highlight=manifest 

I'll remove it.

thanks for the review.

ivantodorovich commented on pos_cashback_warning/static/src/js/screens.js:
```diff
@@ -0,0 +1,47 @@
+        },
+
+    });
+});
```
```suggestion

    return screens;
});
```

Always a good idea to return the object your extending, so that other modules can easily override if needed

ivantodorovich commented on pos_cashback_warning/static/src/js/screens.js:
```diff
@@ -0,0 +1,47 @@
+                    extradue = due;
+                }
+                _.each(lines, function (line) {
+                    console.log("=> line.cashregister.journal.type", line.cashregister.journal.type);
+                    console.log("=> line.amount", line.amount);
```
```suggestion
```

legalsylvain committed:
[FIX] Remove useless console.log calls
[IMP] return screens to improve inheritance
[REF] remove useless key in __manifest__.py file

Co-authored-by: Iván Todorovich <ivan.todorovich@gmail.com>

legalsylvain commented:
Hi @github-actions. Thanks for your friendly reminder ! 
@ivantodorovich : I applied your suggestions.

regards.

legalsylvain commented:
Used in production since 6 monthes without problem.

/ocabot merge patch
